javalib/src/main/scala/java/lang/String.scala

```diff
@@ -667,22 +668,598 @@ final class _String()
+         */
+        def isCased(c: Char): scala.Boolean = {
+          val charType = Character.getType(c)
+          charType == Character.LOWERCASE_LETTER ||
+          charType == Character.UPPERCASE_LETTER ||
+          charType == Character.TITLECASE_LETTER ||
+          ((c >= 0x02B0) && (c <= 0x02B8)) || // Modifier small letter h..y
+          ((c >= 0x02C0) && (c <= 0x02C1)) || // Modifier letter or reversed letter glottal stop
+          ((c >= 0x02E0) && (c <= 0x02E4)) || // Modifier small or reversed small letter gamma
+          c == '\u0345' ||                    // Combining greek ypogegrammeni
+          c == '\u037A' ||                    // Greek ypogegrammeni
+          ((c >= 0x1D2C) && (c <= 0x1D61)) || // Modifier letter capital A to CHI
+          ((c >= 0x2160) && (c <= 0x217F)) || // Roman capital or small numeral one to one thousand
+          ((c >= 0x24B6) && (c <= 0x24E9))    // Circled latin capital or small letters A to Z
+        }
+
+        @tailrec
+        def followsCased(i: Int): scala.Boolean = {
+          if (i < offset) false // Not found non cased in whole string
+          else {
+            val c = charAt(i)
+            // Outside word boundary and not found cased
+            if (c.isWhitespace) false
+            else if (isCased(c)) true
+            else followsCased(i - Character.charCount(c))
+          }
+        }
+
+        @tailrec
+        def precedesOnlyNonCased(i: Int): scala.Boolean = {
+          if (i >= offset + count) true // At string boundary and no cased found
+          else {
+            val c = charAt(i)
+            if (c.isWhitespace) true // At word boundary and no cased found
+            else if (isCased(c)) false
+            else precedesOnlyNonCased(i + Character.charCount(c))
+          }
+        }
+
+        followsCased(idx - 1) && precedesOnlyNonCased(idx + 1)
+      }
+      /* Relevant excerpt from SpecialCasing.txt
+       * # Preserve canonical equivalence for I with dot. Turkic is handled below.
+       *
+       * 0130; 0069 0307; 0130; 0130; # LATIN CAPITAL LETTER I WITH DOT ABOVE
+       * ...
+       * # Special case for final form of sigma
+       *
+       * 03A3; 03C2; 03A3; 03A3; Final_Sigma; # GREEK CAPITAL LETTER SIGMA
+       *
+       * # Note: the following cases for non-final are already in the UnicodeData.txt file.
+       *
+       * # 03A3; 03C3; 03A3; 03A3; # GREEK CAPITAL LETTER SIGMA
+       * # 03C3; 03C3; 03A3; 03A3; # GREEK SMALL LETTER SIGMA
+       * # 03C2; 03C2; 03A3; 03A3; # GREEK SMALL LETTER FINAL SIGMA
+       *
+       * # Note: the following cases are not included, since they would case-fold in lowercasing
+       *
+       * # 03C3; 03C2; 03A3; 03A3; Final_Sigma; # GREEK SMALL LETTER SIGMA
+       * # 03C2; 03C3; 03A3; 03A3; Not_Final_Sigma; # GREEK SMALL LETTER FINAL SIGMA
+
+       */
+      (charAt(i): @switch) match {
+        case '\u03A3' if isFinalCased(i) => "\u03C2"
```

Comment:
Could you call this `if finalSigma(i)`, and quote in the comment of `finalSigma` the definition of `Final_Sigma` in Section 3.13 of the Unicode standard, like is done for `afterI` and `beforeDot`, for example?